### PR TITLE
WIdgets/Button: Fix - Gradient not working with global colors (ED-2292)

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: assets/dev/js/editor/container/container.js [hunk 1]
```diff
@@ -337,4 +337,101 @@ export default class Container extends ArgsObject {
...
+
+			if ( globalValue ) {
+				return globalValue;
+			}
+		}
+
+		return localValue || this.getGlobalDefault( name );
+	}
+
+	getGlobalKey( name ) {
```
Comment (Nevoss) [inline assets/dev/js/editor/container/container.js:360]:
```suggestion
	getGlobalKey( controlName ) {
```

For all the places.

File: assets/dev/js/editor/container/container.js [hunk 2]
```diff
@@ -337,4 +337,101 @@ export default class Container extends ArgsObject {
...
+
+	/**
+	 * Determine if a control's global value is applied.
+	 * It actually checks if the local value is different than the global value.
+	 *
+	 * @param {string} name - Control name
+	 * @returns {boolean}
+	 */
+	isGlobalApplied( name ) {
+		return this.getSetting( name ) !== this.getSetting( name, true );
```
Comment (Nevoss) [inline assets/dev/js/editor/container/container.js:408]:
```suggestion
		return this.settings.get( name ) !== this.getGlobalSetting( name );
```

File: assets/dev/js/editor/container/container.js [hunk 3]
```diff
@@ -337,4 +337,101 @@ export default class Container extends ArgsObject {
 	isDesignable() {
 		return elementor.userCan( 'design' ) && this.isEditable();
 	}
+
+	getSetting( name, localOnly = false ) {
```
Comment (Nevoss) [inline assets/dev/js/editor/container/container.js:341]:
```suggestion
	getGlobalSetting( name ) {
```

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/container/container.js (hunk 1 above)
Comment (StyleShit) [reply assets/dev/js/editor/container/container.js:360]: Done.

File: assets/dev/js/editor/container/container.js [hunk 4]
```diff
@@ -337,4 +337,101 @@ export default class Container extends ArgsObject {
 	isDesignable() {
 		return elementor.userCan( 'design' ) && this.isEditable();
 	}
+
+	getSetting( controlName, localOnly = false ) {
```
Comment (matipojo) [inline assets/dev/js/editor/container/container.js:341]: Explain + tests with 3 scenarios: 
1. Global
2. Local
2. Default Global

**Review by matipojo — CHANGES_REQUESTED** [review]

**Pull request merged**